File: mybatis/src/main/java/data/spring/mybatis/application/required/ProductUseCase.java
package data.spring.mybatis.application.required;

import data.spring.mybatis.adapter.out.persistence.ProductEntity;
import data.spring.mybatis.application.service.command.ProductUpdateCommand;

import java.util.List;

public interface ProductUseCase {
    int saveAll(List<ProductEntity> products);
    int updateAll(List<ProductUpdateCommand> updateBatchRequest);
}

File: src/main/kotlin/data/spring/mybatis/application/provided/product/ProductUseCase.kt
package data.spring.mybatis.application.provided.product

import data.spring.mybatis.application.provided.product.dto.ProductSearchCond
import data.spring.mybatis.application.provided.product.dto.ProductUpdateCommand
import data.spring.mybatis.domain.product.Product
import java.time.LocalDateTime

interface ProductUseCase {
    fun save(product: Product): Int
    fun saveAll(products: List<Product>): Int

    fun update(updateCommand: ProductUpdateCommand): Int
    fun updateAll(updateCommands: List<ProductUpdateCommand>): Int

    fun deleteAll(): Int

    fun findById(productId: Long): Product?
    fun findByCond(searchCond: ProductSearchCond? = null,
                   createdAt: LocalDateTime?,
                   productId: Long?,
                   size: Int = 21): List<Product>
    fun findAll(): List<Product>
}